### ISSUE TITLE
False positive with no-extra-parens

### ISSUE THREAD

**Issue by jr-grenoble** [eslint/eslint#9658]
**Tell us about your environment**

* **ESLint Version:** v4.11.0
* **Node Version:** v8.9.1
* **npm Version:** 5.5.1

**What parser (default, Babel-ESLint, etc.) are you using?**
ECMA2017

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
    "env": {
        "browser": true,
        "node": true,
        "worker": true
    },
    "extends": "eslint:recommended",
    "globals": {
        "Promise": false,
        "Proxy": false,
        "Symbol": false
    },
    "parserOptions": {
        "sourceType": "module",
        "ecmaVersion": 8
    },
    "rules": {
        "for-direction": "warn",
        "getter-return": "error",
        "no-await-in-loop": "warn",
        "no-compare-neg-zero": "warn",
        "no-cond-assign": [
            "error",
            "except-parens"
        ],
        "no-console": "warn",
        "no-constant-condition": "error",
        "no-control-regex": "error",
        "no-debugger": "error",
        "no-dupe-args": "error",
        "no-dupe-keys": "error",
        "no-duplicate-case": "error",
        "no-empty": "error",
        "no-empty-character-class": "error",
        "no-ex-assign": "error",
        "no-extra-boolean-cast": "error",
        "no-extra-parens": [
            "error",
            "all",
            {
                "returnAssign": false
            }
        ],
        "no-extra-semi": "error",
        "no-func-assign": "error",
        "no-inner-declarations": "error",
        "no-irregular-whitespace": [
            "error",
            {
                "skipStrings": true,
                "skipComments": true,
                "skipRegExps": true,
                "skipTemplates": true
            }
        ],
        "no-obj-calls": "error",
        "no-prototype-builtins": "error",
        "indent": [
            "error",
            4
        ],
        "linebreak-style": [
            "error",
            "unix"
        ],
        "quotes": [
            "error",
            "double"
        ],
        "semi": [
            "error",
            "always"
        ]
    }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
"use strict";
export const
    precision = ( operation, x, n ) =>
        n > 0 ?
            ( ( n = 10 ** +n ), Math [ operation ] ( x / n ) * n ) :
            ( ( n = 10 ** -n ), Math [ operation ] (  x * n ) / n );
```

```bash
eslint snippet.js
```

**What did you expect to happen?**
No error nor warning shall be returned: there's no way to remove parens without altering the meaning of the code which is, in each branch of the ternary expression, to assign something to n and then return an expression based on the new value of n. For instance, if you call precision like this:

```js
precision ( "round", 55.48, -1 )
``` 

you expect the second branch of the ternary expression to be equivalent to 

```js
( n = 10 ** 1 ), Math.round ( 55.48 * n ) / n
```
 i.e. to return 55.5.

**What actually happened? Please include the actual, raw output from ESLint.**

/Users/jeanrenebouvier/github/eslint/snippet.js
  5:15  error  Gratuitous parentheses around expression  no-extra-parens
  6:15  error  Gratuitous parentheses around expression  no-extra-parens

✖ 2 problems (2 errors, 0 warnings)
  2 errors, 0 warnings potentially fixable with the `--fix` option.

**Comment by not-an-aardvark** [comment]
I think the inner set of parentheses can be removed without changing the semantics:

```js
"use strict";
export const
    precision = ( operation, x, n ) =>
        n > 0 ?
            ( n = 10 ** +n, Math [ operation ] ( x / n ) * n ) :
            ( n = 10 ** -n, Math [ operation ] (  x * n ) / n );
```

**Comment by jr-grenoble** [comment]
Thanks Teddy! My apologies for the extra work. - jr


<https://icon.me/jeanrene?from=emailsignature&access=65b8b3e1677d51e77b65c69e8bd3fdc30ecbebb6>
Jeanrene Bouvier
Head of Facts Haven
jeanrene.bouvier@gmail.com
View my Icon
<https://icon.me/jeanrene?from=emailsignature&access=65b8b3e1677d51e77b65c69e8bd3fdc30ecbebb6>

**Comment by not-an-aardvark** [comment]
No worries. Closing this issue since it's not a bug.
